### DISCUSSION TITLE
Firmware with double precision for esp32

### DISCUSSION THREAD

**Discussion by priisdk** [micropython/micropython#9670] [General]
For some astronomical calculations I need double precision, and I have found some firmware files with dp here:
https://gitlab.com/rcolistete/micropython-firmwares/-/commit/5a807f7506bc7452a154867c6b130981afd58368#60a723e4ad3bc25ec8be7c508d17d4836d326b32
However, I also need urllib, urllib.request,  ntptime and SoftI2C (from machine), and I cannot make it work together with any of the mentioned firmware. In some cases I cannot install urllib, in other cases there is no ntptime or SoftI2C.
Where can I find an up-to-date fully working firmware with double precision for my esp32S (Nodemcu, wifiESP-WROOM-32)?
Or should I buy a better microcomputer? If yes, which one?

Poul Riis

**Comment by jimmo** [comment]
> Where can I find an up-to-date fully working firmware with double precision for my esp32S (Nodemcu, wifiESP-WROOM-32)?

I'm not aware of any published firmware. But you can compile it yourself by following the instructions:
https://github.com/micropython/micropython/tree/master/ports/esp32#micropython-port-to-the-esp32
(ask here if you have questions)

To enable double-precision, in `ports/esp32/mpconfigport.h` change

```c
#define MICROPY_FLOAT_IMPL                  (MICROPY_FLOAT_IMPL_FLOAT)
```

to

```c
#define MICROPY_FLOAT_IMPL                  (MICROPY_FLOAT_IMPL_DOUBLE)
```
